neo/idlib/math/Simd_AVX.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+===========================================================================
+
+Doom 3 GPL Source Code
+Copyright (C) 1999-2011 id Software LLC, a ZeniMax Media company.
+
+This file is part of the Doom 3 GPL Source Code ("Doom 3 Source Code").
+
+Doom 3 Source Code is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Doom 3 Source Code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Doom 3 Source Code.  If not, see <http://www.gnu.org/licenses/>.
+
+In addition, the Doom 3 Source Code is also subject to certain additional terms. You should have received a copy of these additional terms immediately following the terms and conditions of the GNU General Public License which accompanied the Doom 3 Source Code.  If not, please request a copy in writing from id Software at the address below.
+
+If you have questions concerning this license or the applicable additional terms, you may contact in writing id Software LLC, c/o ZeniMax Media Inc., Suite 120, Rockville, Maryland 20850 USA.
+
+===========================================================================
+*/
+
+#include "sys/platform.h"
+#include "Simd_AVX.h"
+
+//===============================================================
+//
+//	AVX implementation of idSIMDProcessor
+//
+//===============================================================
+
+#if defined(__GNUC__) && defined(__SSE3__)
+
+/*
+============
+idSIMD_SSE3::GetName
+============
+*/
+const char *idSIMD_AVX::GetName( void ) const {
+	return "MMX & SSE & SSE2 & SSE3 & AVX";
+}
+
+#elif defined(_MSC_VER) && defined(_M_IX86)
```

Comment:
why is this code (and the AVX2 code) only enabled when using MSVC on 32bit(!) x86?